go.mod

```diff
@@ -1,6 +1,8 @@
 module terraform-provider-maas
 
-go 1.22.7
+go 1.23
```

Comment:
If we are going to use go-set/v3, we also need to bump to Go 1.23 without including the toolchain.